src/main/java/com/easysplit/ess/transactions/application/TransactionsServiceImpl.java

```diff
@@ -38,4 +38,11 @@ public Transaction createNewTransaction(Transaction transaction) {
 
         return createdTransaction.toTransaction();
     }
+
+    public Transaction getTransaction(String transactionGuid){
+        TransactionEntity transaction = transactionsRepository.getTransaction(transactionGuid);
+
```

Comment:
Here we should have one more line, something like `DebtEntity debtEntity = transactionsReposity.getDebt(transactionGuid);`

After that we should set the debt `trasanction.setDebt(debtEntity)`

Reply:
So the DebtEntity is supposed to be retrieved through the transaction id?